api/resources_portal/models/notification.py

```diff
@@ -0,0 +1,86 @@
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from computedfields.models import ComputedFieldsModel, computed
+
+from resources_portal.models.material import Material
+from resources_portal.models.organization import Organization
+from resources_portal.models.user import User
+
+
+class Notification(ComputedFieldsModel):
+    class Meta:
+        db_table = "notifications"
+        get_latest_by = "created_at"
+
+    NOTIFICATION_TYPES = (
+        ("ORG_REQUEST_CREATED", "ORG_REQUEST_CREATED"),
+        ("ORG_INVITE_CREATED", "ORG_INVITE_CREATED"),
+        ("ORG_INVITE_ACCEPTED", "ORG_INVITE_ACCEPTED"),
+        ("ORG_REQUEST_ACCEPTED", "ORG_REQUEST_ACCEPTED"),
+        ("ORG_INVITE_REJECTED", "ORG_INVITE_REJECTED"),
+        ("ORG_REQUEST_REJECTED", "ORG_REQUEST_REJECTED"),
+        ("ORG_INVITE_INVALID", "ORG_INVITE_INVALID"),
+        ("ORG_REQUEST_INVALID", "ORG_REQUEST_INVALID"),
+        ("MTA_UPLOADED", "MTA_UPLOADED"),
+        ("APPROVE_REQUESTS_PERM_GRANTED", "APPROVE_REQUESTS_PERM_GRANTED"),
+        ("TRANSFER_REQUESTED", "TRANSFER_REQUESTED"),
+    )
+
+    objects = models.Manager()
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    notification_type = models.CharField(max_length=32, choices=NOTIFICATION_TYPES)
+    notified_user = models.ForeignKey(
+        User, blank=False, null=False, on_delete=models.CASCADE, related_name="notifications"
+    )
+    associated_user = models.ForeignKey(
+        User,
+        blank=False,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="associated_notifications",
+    )
+    associated_organization = models.ForeignKey(
+        Organization, blank=False, null=True, on_delete=models.CASCADE
+    )
+    associated_material = models.ForeignKey(
+        Material, blank=False, null=True, on_delete=models.CASCADE
+    )
+
+    @computed(models.EmailField(blank=False, null=False))
+    def email(self):
+        return self.notified_user.email
```

Comment:
Oh I see, you added it here. Can we not do this as a computed field and instead set it when we send the notification so it cannot be changed by the user?